src/txn/txn_rollback_to_stable.c

```diff
@@ -137,6 +71,198 @@ __txn_abort_newer_insert(
             __txn_abort_newer_update(session, ins->upd, rollback_timestamp);
 }
 
+/*
+ * __txn_abort_row_replace_upd_list --
+ *     Abort updates in the update list and replace it with the provided update
+ *
+ */
+static inline int
+__txn_abort_row_replace_upd_list(
+  WT_SESSION_IMPL *session, WT_PAGE *page, WT_ROW *rip, WT_UPDATE *upd)
+{
+    WT_DECL_RET;
+    WT_PAGE_MODIFY *mod;
+    WT_UPDATE *old_upd, **upd_entry;
+    size_t upd_size;
+
+    /* If we don't yet have a modify structure, we'll need one. */
+    WT_RET(__wt_page_modify_init(session, page));
+    mod = page->modify;
+
+    /* Allocate an update array as necessary. */
+    WT_PAGE_ALLOC_AND_SWAP(session, page, mod->mod_row_update, upd_entry, page->entries);
+
+    /* Set the WT_UPDATE array reference. */
+    upd_entry = &mod->mod_row_update[WT_ROW_SLOT(page, rip)];
+    upd_size = __wt_update_list_memsize(upd);
+
+    /*
+     * If it's a full update list, we're trying to instantiate the row. Otherwise, it's just a
+     * single update that we'd like to append to the update list.
+     *
+     * Set the "old" entry to the second update in the list so that the serialization function
+     * succeeds in swapping the first update into place.
+     */
+    if (upd->next != NULL)
+        *upd_entry = upd->next;
+    old_upd = *upd_entry;
+
+    /*
+     * Point the new WT_UPDATE item to the next element in the list. If we get it right, the
+     * serialization function lock acts as our memory barrier to flush this write.
```

Comment:
I suggest removing 'If we get it right,' because that sounds like we hope we're lucky and maybe we won't be lucky.
Also remove the word 'lock' as there is no lock being used here.

Reply:
Fixed.